src/main/java/dev/j3fftw/worldeditslimefun/commands/WorldEditSlimefunCommands.java

```diff
@@ -1,115 +1,187 @@
+import co.aikar.commands.BukkitCommandExecutionContext;
+import co.aikar.commands.CommandCompletions;
+import co.aikar.commands.CommandContexts;
+import co.aikar.commands.PaperCommandManager;
 import co.aikar.commands.annotation.CommandAlias;
 import co.aikar.commands.annotation.CommandCompletion;
 import co.aikar.commands.annotation.CommandPermission;
 import co.aikar.commands.annotation.Default;
 import co.aikar.commands.annotation.Subcommand;
 import dev.j3fftw.worldeditslimefun.WorldEditSlimefun;
-import io.github.thebusybiscuit.slimefun4.api.items.ItemHandler;
+import dev.j3fftw.worldeditslimefun.commands.flags.CommandFlag;
+import dev.j3fftw.worldeditslimefun.commands.flags.CommandFlags;
+import dev.j3fftw.worldeditslimefun.utils.PositionManager;
+import dev.j3fftw.worldeditslimefun.utils.Utils;
 import io.github.thebusybiscuit.slimefun4.api.items.SlimefunItem;
 import io.github.thebusybiscuit.slimefun4.core.handlers.BlockBreakHandler;
 import io.github.thebusybiscuit.slimefun4.core.handlers.BlockPlaceHandler;
 import io.github.thebusybiscuit.slimefun4.implementation.items.blocks.UnplaceableBlock;
 import io.github.thebusybiscuit.slimefun4.libraries.dough.blocks.BlockPosition;
 import me.mrCookieSlime.Slimefun.api.BlockStorage;
 import org.bukkit.ChatColor;
-import org.bukkit.Location;
 import org.bukkit.Material;
 import org.bukkit.block.Block;
 import org.bukkit.block.BlockFace;
 import org.bukkit.entity.Player;
 import org.bukkit.event.block.BlockBreakEvent;
 import org.bukkit.event.block.BlockPlaceEvent;
+import org.bukkit.inventory.EquipmentSlot;
+import org.bukkit.inventory.ItemStack;
 
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.function.Consumer;
+
+@SuppressWarnings("unused")
 @CommandAlias("wesf|sfedit")
+@CommandPermission("wesf.admin")
 public class WorldEditSlimefunCommands extends BaseCommand {
 
+    public static void init(WorldEditSlimefun plugin) {
+        PaperCommandManager manager = new PaperCommandManager(plugin);
+        CommandCompletions<BukkitCommandCompletionContext> completions = manager.getCommandCompletions();
+        CommandContexts<BukkitCommandExecutionContext> contexts = manager.getCommandContexts();
+
+        completions.registerStaticCompletion("slimefun_blocks", Utils.SLIMEFUN_BLOCKS);
+        completions.registerAsyncCompletion("command_flags", context -> {
+            List<String> args = new ArrayList<>(Arrays.asList(context.getContextValueByName(String[].class, "commandFlags")));
+            List<String> availableFlags = new ArrayList<>(CommandFlags.FLAG_TYPES.keySet());
+            availableFlags.removeAll(args);
+
+            if (args.isEmpty()) {
+                return availableFlags;
+            }
+
+            String currentArg = args.remove(args.size() - 1);
+            if (args.isEmpty()) {
+                return availableFlags;
+            }
+
+            String lastArg = args.get(args.size() - 1);
```

Comment:
shouldnt this use currentArg?

Reply:
no....?

Reply:
oh I guess I never actually use currentArg do I, lol.

I can probably remove the variable declaration then? But it might make it even less readable doing so

currentArg is the argument actively being typed. acf will handle filtering the options based on the current argument so leaving it alone is all we need to do. What's really important is knowing what the last full argument was.